### PR TITLE
Upgrade to onnxruntime 0.19

### PR DESCRIPTION
**Pull request by yeldarby**
# Description

This pulls up our dependencies the the latest onnxruntime and updates our Docker containers to match.

Benefits:
* Gets MPS acceleration on Mac since onnxruntime-silicon is now mainlined into onnxruntime
* Unlocks upgrade to CUDA 12 and newer NVIDIA compute capabilities (#181)
* Preps for adding Jetpack 6 support to our Jetson containers

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Have run the tests locally. About to build all the containers via our github actions.

## Any specific deployment considerations

No.

## Docs

- Not needed

### CODE REVIEW

File: docker/dockerfiles/Dockerfile.onnx.gpu [hunk 1]
```diff
@@ -28,7 +28,9 @@ COPY requirements/requirements.sam.txt \
     ./
 
 RUN python3 -m pip install -U pip
-RUN python3 -m pip install --extra-index-url https://download.pytorch.org/whl/cu118 \
+RUN python3 -m pip install \
+    --extra-index-url https://download.pytorch.org/whl/cu118 \
+    --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-11/pypi/simple/ \
```
Comment (probicheaux) [inline docker/dockerfiles/Dockerfile.onnx.gpu:33]: what is this guy doing? changing the default cuda version for onnxruntime? so onnxruntime installs cuda?

File: inference/core/env.py [hunk 2]
```diff
@@ -264,7 +264,7 @@
 
 ONNXRUNTIME_EXECUTION_PROVIDERS = os.getenv(
     "ONNXRUNTIME_EXECUTION_PROVIDERS",
-    "[CUDAExecutionProvider,OpenVINOExecutionProvider,CPUExecutionProvider]",
+    "[CUDAExecutionProvider,OpenVINOExecutionProvider,CoreMLExecutionProvider,CPUExecutionProvider]",
```
Comment (probicheaux) [inline inference/core/env.py:267]: now everyone not on a mac is gonna get stupid warnings about not having coremlexecution provider :)

File: requirements/requirements.cpu.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
-onnxruntime~=1.15.1
```
Comment (probicheaux) [inline requirements/requirements.cpu.txt:1]: `>=1.15.1`

**Review by probicheaux — APPROVED** [review]

File: docker/dockerfiles/Dockerfile.onnx.gpu (hunk 1 above)
Comment (yeldarby) [reply docker/dockerfiles/Dockerfile.onnx.gpu:33]: The default is now to install the onnxruntime that's compatible with CUDA 12; this forces it to install the one built for CUDA 11 instead. (It used to be the reverse; you had to choose the channel for CUDA 12 and the default was 11)

File: inference/core/env.py (hunk 2 above)
Comment (yeldarby) [reply inference/core/env.py:267]: Yeh I was thinking we should probably update this default to be what `onnxruntime.get_available_providers()` reports but was going to do that in a different PR.

File: requirements/requirements.cpu.txt (hunk 3 above)
Comment (yeldarby) [reply requirements/requirements.cpu.txt:1]: Went with `onnxruntime>=1.15.1,<1.20.0`

**Comment by yeldarby** [comment]
Update: tests have all passed & containers build.

image

image

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**